python/cudf/cudf/core/dataframe.py

```diff
@@ -2662,6 +2662,8 @@ def apply_rows(
             result of applying func against the underlying column data, which
             may be garbage.
```

Comment:
Instead of duplicating this for apply_rows and apply_chunks, we should move it into the applyutils `{params}` variable so it auto populates both.

Reply:
So apparently I need to get brushed up on how our documentation generation works. What are we using?

Reply:
I didn't realize this was a thing. I'm also moving the `df` parameter doc in there as well, since it's shared.

IMO a doc util like this is a code smell, though. I realize having two comments nearly identical introduces some tech debt in the form of deviation, and that the util solves that particular problem, but it begs the question... why are these functions so similar in signature?

`cache_key`, `chunks`, and `tpb` are the only non-uniform parameters, and I'd presuppose that will remain the case, since even those parameters are tightly coupled (cant cache if using chunks/tpb, for instance - or maybe the cache key just changes?).

Should we look at consolidating these functions to deprecate `apply_chunks` in favor of an "overloaded" `apply_rows`, or just leave it as is? Mainly asking because I personally don't want to set precedence for variables in our documentation (I'm open to differing opinions, though).

Reply:
What I'm really curious about is why `apply_rows` and `apply_chunks` are distinct functions, and is that really the desired API? Can we just key it off the presence of the "chunks" argument?

Reply:
Defer to you guys on that one, but from my perspective I think unifying them is appealing longterm (as long as `apply_chunks` is preserved as an alias for a while so existing code doesn't break)